File: LivrusVirtus/wxWindows/wx.NET/Delphi.NET/fontes/CadAssuntos.pas
unit CadAssuntos;

interface

uses
  System.Drawing, wx, Cadastros;

type
  TfrmCadAssuntos = class(TfrmCadastros)
  private
    function ValidaDados(bTodosDados: boolean): boolean;
    procedure OnClose(sender: System.Object; e: wx.Event);
    procedure OnBtnNovo(sender: System.Object; e: wx.Event);
    procedure OnBtnPesquisar(sender: System.Object; e: wx.Event);
    procedure OnBtnAnterior(sender: System.Object; e: wx.Event);
    procedure OnBtnProximo(sender: System.Object; e: wx.Event);
    procedure OnBtnSalvar(sender: System.Object; e: wx.Event);
    procedure OnBtnExcluir(sender: System.Object; e: wx.Event);
    procedure OnBtnLimpar(sender: System.Object; e: wx.Event);
    procedure OnBtnFechar(sender: System.Object; e: wx.Event);
  public
    constructor Create(parent: wx.Window);
  end;

var
  frmCadAssuntos: TfrmCadAssuntos;

  [assembly: RuntimeRequiredAttribute(TypeOf(TfrmCadAssuntos))]

implementation

uses Login, TextValidatorCtrl, RotinasGlobais, Resource, ConsultasSQL;

constructor TfrmCadAssuntos.Create(parent: wx.Window);
begin
  inherited Create(parent, FILTER_NUMERIC, true, 'Código:', 'Assunto:');

  //labDescricao.Label:='Assunto: ';

  EVT_BUTTON(ID_FRMCADASTROS_BTNNOVO, OnBtnNovo );
  EVT_BUTTON(ID_FRMCADASTROS_BTNPESQUISAR, OnBtnPesquisar );
  EVT_BUTTON(ID_FRMCADASTROS_BTNANTERIOR, OnBtnAnterior );
  EVT_BUTTON(ID_FRMCADASTROS_BTNPROXIMO, OnBtnProximo );
  EVT_BUTTON(ID_FRMCADASTROS_BTNSALVAR, OnBtnSalvar );
  EVT_BUTTON(ID_FRMCADASTROS_BTNEXCLUIR, OnBtnExcluir );
  EVT_BUTTON(ID_FRMCADASTROS_BTNLIMPAR, OnBtnLimpar );
  EVT_BUTTON(ID_FRMCADASTROS_BTNFECHAR, OnBtnFechar );
	EVT_CLOSE( OnClose );
end;

function TfrmCadAssuntos.ValidaDados(bTodosDados: boolean): boolean;
begin
	result:=true;
  if Rotinas.ValidaCampo(edtCodigo.Value, Res.STR_CODNINF) then
    result:=false;
  if bTodosDados then begin
    if Rotinas.ValidaCampo(edtDescricao.Value, Res.STR_ASSNINF) then
    	result:=false;
  end;
end;

procedure TfrmCadAssuntos.OnClose(sender: System.Object; e: wx.Event);
begin
  self.Parent.Close();
	e.Skip();
end;

procedure TfrmCadAssuntos.OnBtnNovo(sender: System.Object; e: wx.Event);
begin
  btnNovoClick();
  NovoDado(ConsSQL.Assunto('N', edtCodigo.Value,''));	
  e.Skip();
end;

procedure TfrmCadAssuntos.OnBtnPesquisar(sender: System.Object; e: wx.Event);
begin
  PesquisarDados(ConsSQL.Assunto('S', edtCodigo.Value,''), 
    Res.STR_ASSNENC);
  e.Skip();
end;

procedure TfrmCadAssuntos.OnBtnAnterior(sender: System.Object; e: wx.Event);
begin
  registroAnterior();	
  e.Skip();
end;

procedure TfrmCadAssuntos.OnBtnProximo(sender: System.Object; e: wx.Event);
begin
  registroProximo();		
  e.Skip();
end;

procedure TfrmCadAssuntos.OnBtnSalvar(sender: System.Object; e: wx.Event);
begin
  SalvarDados(ValidaDados(true),
    ConsSQL.Assunto('S', edtCodigo.Value,''),
    ConsSQL.Assunto('U', edtCodigo.Value, edtDescricao.Value),
    ConsSQL.Assunto('I', edtCodigo.Value, edtDescricao.Value), true);
  e.Skip();
end;

procedure TfrmCadAssuntos.OnBtnExcluir(sender: System.Object; e: wx.Event);
begin
  ExcluirDados(ValidaDados(false),
    ConsSQL.Assunto('D', edtCodigo.Value,''));
  e.Skip();
end;

procedure TfrmCadAssuntos.OnBtnFechar(sender: System.Object; e: wx.Event);
begin
  self.Close(true);
  e.Skip();
end;

procedure TfrmCadAssuntos.OnBtnLimpar(sender: System.Object; e: wx.Event);
begin
  btnLimparClick();
  e.Skip();
end;

end.

File: LivrusVirtus/KDevelop/wxWin/ODBC/Motif/plivrus/CadAssuntos.cpp
#include "resource.h"
#include "Cadastros.h"
#include "CadAssuntos.h"
#include "Principal.h"
#include "ConsultasSQL.h"

BEGIN_EVENT_TABLE(CfrmCadAssuntos, CfrmCadastros)
  EVT_BUTTON(ID_FRMCADASTROS_BTNNOVO, CfrmCadAssuntos::OnBtnNovo)
  EVT_BUTTON(ID_FRMCADASTROS_BTNPESQUISAR, CfrmCadAssuntos::OnBtnPesquisar)
  EVT_BUTTON(ID_FRMCADASTROS_BTNANTERIOR, CfrmCadAssuntos::OnBtnAnterior)
  EVT_BUTTON(ID_FRMCADASTROS_BTNPROXIMO, CfrmCadAssuntos::OnBtnProximo)
  EVT_BUTTON(ID_FRMCADASTROS_BTNSALVAR, CfrmCadAssuntos::OnBtnSalvar)
  EVT_BUTTON(ID_FRMCADASTROS_BTNEXCLUIR, CfrmCadAssuntos::OnBtnExcluir)
  EVT_BUTTON(ID_FRMCADASTROS_BTNLIMPAR, CfrmCadAssuntos::OnBtnLimpar)
  EVT_BUTTON(ID_FRMCADASTROS_BTNFECHAR, CfrmCadAssuntos::OnBtnFechar)
  EVT_CLOSE(CfrmCadAssuntos::OnClose)
END_EVENT_TABLE()

CfrmCadAssuntos::CfrmCadAssuntos(wxWindow *parent)
       : CfrmCadastros(parent, wxFILTER_NUMERIC, true) {
  CfrmCadAssuntos::IniciaControles();
}

void CfrmCadAssuntos::IniciaControles() {
 this->labDescricao->SetLabel(wxT("Assunto: "));
}

CfrmCadAssuntos::~CfrmCadAssuntos() { }

bool CfrmCadAssuntos::ValidaDados(bool bTodosDados)
{
if (Rotinas->ValidaCampo(edtCodigo->GetValue(), STR_CODNINF))
  	  return false;
  	if (bTodosDados) {
  	  if (Rotinas->ValidaCampo(edtDescricao->GetValue(), STR_ASSNINF))
  		return false;
  	}
  	return true;
}

void CfrmCadAssuntos::OnClose(wxCloseEvent& event) {
  this->GetParent()->Close();
  event.Skip();
}

void CfrmCadAssuntos::OnBtnNovo(wxCommandEvent& event) {
  btnNovoClick();
  NovoDado(ConsSQL->Assunto('N', edtCodigo->GetValue(),""));	
  event.Skip();
}

void CfrmCadAssuntos::OnBtnPesquisar(wxCommandEvent& event) {
  PesquisarDados(
    ConsSQL->Assunto('S', edtCodigo->GetValue(),""), STR_ASSNENC);
  event.Skip();
}

void CfrmCadAssuntos::OnBtnAnterior(wxCommandEvent& event) {
  registroAnterior();	
  event.Skip();
}

void CfrmCadAssuntos::OnBtnProximo(wxCommandEvent& event) {
  registroProximo();		
  event.Skip();
}

void CfrmCadAssuntos::OnBtnSalvar(wxCommandEvent& event) {
  SalvarDados(ValidaDados(true),
    ConsSQL->Assunto('S', edtCodigo->GetValue(),""),
    ConsSQL->Assunto('U', edtCodigo->GetValue(), edtDescricao->GetValue()),
    ConsSQL->Assunto('I', edtCodigo->GetValue(), edtDescricao->GetValue()), 
    true);
  event.Skip();
}

void CfrmCadAssuntos::OnBtnExcluir(wxCommandEvent& event) {
  ExcluirDados(ValidaDados(false),
    ConsSQL->Assunto('D', edtCodigo->GetValue(),""));
  event.Skip();
}

void CfrmCadAssuntos::OnBtnFechar(wxCommandEvent& event) {
  this->Close(true);
  event.Skip();
}

void CfrmCadAssuntos::OnBtnLimpar(wxCommandEvent& event) {
  btnLimparClick();
  event.Skip();
}

